src/vs/workbench/contrib/chat/common/promptSyntax/service/promptsServiceImpl.ts

```diff
@@ -152,12 +156,168 @@ export class PromptsService extends Disposable implements IPromptsService {
 		const prompts = await Promise.all([
 			this.fileLocator.listFiles(type, PromptsStorage.user, token).then(uris => uris.map(uri => ({ uri, storage: PromptsStorage.user, type } satisfies IUserPromptPath))),
 			this.fileLocator.listFiles(type, PromptsStorage.local, token).then(uris => uris.map(uri => ({ uri, storage: PromptsStorage.local, type } satisfies ILocalPromptPath))),
-			this.getExtensionContributions(type)
+			this.getExtensionContributions(type),
+			type === PromptsType.agent ? this.listCustomAgentsFromProvider(token) : []
 		]);
 
 		return [...prompts.flat()];
 	}
 
+	/**
+	 * Registry of CustomAgentsProvider instances. Extensions can register providers via the proposed API.
+	 */
+	private readonly customAgentsProviders: Array<{
+		extension: IExtensionDescription;
+		provideCustomAgents: (options: ICustomAgentQueryOptions, token: CancellationToken) => Promise<IExternalCustomAgent[] | undefined>;
+	}> = [];
+
+	/**
+	 * Registers a CustomAgentsProvider. This will be called by the extension host bridge when
+	 * an extension registers a provider via vscode.chat.registerCustomAgentsProvider().
+	 */
+	public registerCustomAgentsProvider(extension: IExtensionDescription, provider: {
+		provideCustomAgents: (options: ICustomAgentQueryOptions, token: CancellationToken) => Promise<IExternalCustomAgent[] | undefined>;
+	}): IDisposable {
+		if (!isProposedApiEnabled(extension, 'chatParticipantPrivate')) {
```

Comment:
add a constant

Reply:
do you mean for `'chatParticipantPrivate'`? I see that other files just use this literal directly as well